### ISSUE TITLE
Element Width/Height Animation

### ISSUE THREAD

**Issue by sassomedia** [amcharts/amcharts5#61]
I seem to have issues with animating `width`/`height` of an element (`Rectangle`, `Container`, etc.). I tried a number of different scenarios in this [pen](https://codepen.io/sassomedia/pen/mdMjLjW?editors=0010). Here's the overview:

If I animate a property like `fill`, this works as expected and the color changes:
```javascript
const testContainer = am5.Container.new(root, { x: 0, y: 0, width: am5.p100, height: am5.p100 });
const testBg = am5.Rectangle.new(root, { x: 0, y: 0, width: am5.p50, height: am5.p50, fill: am5.color(0xff0000) });

testContainer.children.push(testBg);
series.set("background", testContainer);

testBg.animate({
  key: "fill",
  to: am5.color(0xff00ff),
  duration: 500,
  easing: am5.ease.out(am5.ease.cubic)
});
```
1. If I try this with `width`, the following does not work:
```javascript
testBg.animate({
  key: "width",
  // from: am5.p50, // #)
  to: am5.p100,
  duration: 500,
  easing: am5.ease.out(am5.ease.cubic)
});
```
#) There also seems to be a bug in `Animation.ts` when it tries to animate this as the current `width` retrieved will be in pixel as opposed to be in percent which the element was originally set up with. There is [no condition handling two different type of values](https://github.com/amcharts/amcharts5/blob/master/src/.internal/core/util/Animation.ts#L99-L113) to animate from - to. However, even if `from` were supplied in percent as well, it does not work.

2. So I tried with pixel animation but that did not work either:
```javascript
testBg.animate({
  key: "width",
  to: 1000,
  duration: 500,
  easing: am5.ease.out(am5.ease.cubic)
});
```

3. Since `Rectangle` inherits `width` from `Sprite`, I figured maybe I would need to animate the sprite but I don't seem to get access to that as `testBg.get("sprite")` returns `undefined`.

4. Also tried just setting the `width` without any animation and that was also unsuccessful:
```javascript
testBg.set("width", am5.p100);
```

5. Inspecting the `Rectangle`, I found an `animatePrivate` function and gave that a try. No go either:
```javascript
testBg.animatePrivate({
  key: "width",
  // from: am5.p50, // even with "from" supplied in percent, it does not work
  to: am5.p100,
  duration: 500,
  easing: am5.ease.out(am5.ease.cubic)
});
```

6. Finally, using the `animatePrivate` function with pixel did the trick and it animated the width.
```javascript
testBg.animatePrivate({
  key: "width",
  to: 1000,
  duration: 500,
  easing: am5.ease.out(am5.ease.cubic)
});
```
However none of this is in the docs so I don't think that's the right way to do it. I would also still need the percent animation to work.

What would be the right way to achieve this animation? TIA!

**Comment by zeroin** [comment]
We don't plan to support animation between different types, sorry. 
As for your case - width and height of the background of a Container is set by the Container, so all your animations are reset by this. I would recommend to add a Rectangle as a child to the Container and animate it.

**Comment by sassomedia** [comment]
@zeroin 
> We don't plan to support animation between different types, sorry.

Yeah, at some point I seem to have gotten mixed types but I'm not seeing this anymore, so that might not be an issue.

> As for your case - width and height of the background of a Container is set by the Container, so all your animations are reset by this. I would recommend to add a Rectangle as a child to the Container and animate it.

Are you referring to the container being set as a "background" of the Series?
```javascript
series.set("background", testContainer);
```
Even if I instead add the container to the chart's container, it still does not animate the `width`:
```javascript
chart.plotContainer.children.push(testContainer);
```
Nor does setting the `width` without animation work when in that container:
```javascript
testBg.set("width", am5.p100);
```

Are there any other suggestions I could try? Thanks. Updated pen [here](https://codepen.io/sassomedia/pen/mdMjLjW?editors=0010).

**Comment by martynasma** [comment]
Fixed in [5.0.5](https://github.com/amcharts/amcharts5/releases/tag/5.0.5).

## [5.0.5] - 2021-11-15

### Added
- New root element method: `moveDOM()` (accepts element id or reference as a parameter). Can be used to move whole chart to some other container within DOM.

### Fixed
- `ExportingMenu` was interfering with functionality of form elements on the same page (e.g. select).
- Fixed issues with text kerning with styling blocks mid-word.
- Animating values of `width`/`height` settings using percent values was not working properly.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download-v5/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.

**Comment by sassomedia** [comment]
Confirmed, working now as expected! Thanks! 🙏
